standards/naming.md

```diff
@@ -118,56 +118,119 @@ collectedItem: []
         "authorName": "John Doe",               // the parent entity is an author, so it should be "name" for the attribute name.
         "publisherName": "My Publishing House"  // this name attribute MUST have a prefix for another entity, since its not the name of this entity "author"
-        "agentId": 2                            // this name is appropriate with Id following its associated entity name.
     }
 }
  
 // CORRECT EXAMPLE
-GET /v1/books/2345
+GET /v1/books/5196ab21
 RESPONSE
 {
-    "id": 2345,                            
+    "id": "5196ab21",                            
     "name": "The Best Book",
     "author": {                    
-        "id":456,                      
+        "id": "3793213e",                      
         "name": "John Doe",            
-        "publisherName": "My Publishing House",
-        "agentId": 2
+        "publisherName": "My Publishing House"
     }
 }
 ```
 
+## Identifiers
+
+- An entity with a unique identifier **MUST** be labeled as `id`.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",                            
+    "name": "The Best Book"
+}
+```
+
+- Any unique identifier of an entity **MUST** have Id after the entity name when referenced from another entity.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",
+    "name": "The Best Book",
+    "authorId": "3793213e"      // usage of "authorId" references this is the id for a different entity.
+}
+```
+
+- Unique identifiers **SHOULD** be string values and considered opaque as described in [serialization of numbers](serialization.md#number).
+
+```
+// INCORRECT
+{
+    "id": 2
+}
+
+// CORRECT
+{
+    "id": "b96cb3ead9a9"
+}
+```
+
+## Domain References
+
+- Unique references to other significant entities within your API Domain, but in a different root resource **SHOULD** use [URN (Uniform Resource Name)](https://en.wikipedia.org/wiki/Uniform_Resource_Name)-like references where appropriate.  
+    - URN-like refers to using the convention of specifying agreed upon namespaces, entities followed by the object identifier, but without the `urn` prefix or formal registration of the namespace ID with IANA according to [RFC8141](https://datatracker.ietf.org/doc/rfc8141/). This provides interoperability and durability benefits within your API and endpoint ecosystem.
+    - URN-like references **MUST** be the form of standard `URNs` without the `urn:` prefix to avoid confusion of official registration. A simplified example would be: `{namespace}:{entity}:{id}` (e.g. `spsc:book:b96cb3ead9a9`).
+        - URN-like references **MUST** have a max-length of 255 characters.
+        - URN-like references **MUST** be case-sensitive.
+        - URN-like `{namespace}` **MUST** be a single value applied across all API endpoint response for the usage of self-referencing `ref` properties. At SPS Commerce this value **MUST** be `spsc`.
```

Comment:
Just pointing out that we already use another acronym in another section of [API standards](https://github.com/SPSCommerce/sps-api-standards/blob/main/standards/request-response.md#custom-headers) - `sps`. Is there any intention here, or maybe we have some documented rules regarding choosing the right one?

Reply:
The rule for usage at SPS Commerce is that you MUST use the one indicated here currently as `spsc`.

BUT I love that you call out for headers we have `sps` already. Thats an ugly little difference. Obviously given this one is in flight, its easy to make URNs "sps" now.... thanks for identifying this inconsistency.

@alexander-ivakhnenko how strong is your opinion on keeping the namespace "spsc" vs "sps" here?

Reply:
This is now updated to be "sps" to ensure high-level consistency. Thanks for pointing this out!!!